### PR TITLE
[JCLOUDS-1468] Adding missing options for the ListUsageRecords API call.

### PR DESCRIPTION
**Pull request by mcdonnell-john**
related to https://issues.apache.org/jira/browse/JCLOUDS-1468

### CODE REVIEW

**Comment by andreaturli** [comment]
Thanks @mcdonnell-john for this. is there a jira issue tracking this problem, could you maybe improve the description pointing to that jira issue?

Also have you signed http://apache.org/licenses/icla.pdf as per https://www.apache.org/foundation/how-it-works/legal.html#incoming-code ? Thanks again

**Comment by mcdonnell-john** [comment]
Hi @andreaturli 

Yep, I created https://issues.apache.org/jira/browse/JCLOUDS-1468 for this PR, I could probably look to add more info there - will look to do so later/tomorrow.

Also I'm part of the member of the Apache Netbeans Incubating project so I have an apache account/id.

**Comment by mcdonnell-john** [comment]
@andreaturli  apologies for the delay, I've updated the JIRA ticket now: https://issues.apache.org/jira/browse/JCLOUDS-1468

File: apis/cloudstack/src/main/java/org/jclouds/cloudstack/domain/UsageRecord.java [hunk 1]
```diff
@@ -400,6 +510,15 @@ protected UsageRecord(String id, @Nullable String description, @Nullable String
...
       this.project = project;
       this.projectId = projectId;
+      this.domain = domain;
+      this.virtualSize = virtualSize;
+      this.cpuNumber = cpuNumber;
+      this.cpuSpeed = cpuSpeed;
+      this.memory = memory;
+      this.isSystem = isSystem;
+      this.networkId = networkId;
+      this.isDefault = isDefault;
```
[the remark this answers is not in the document]
Comment (mcdonnell-john) [reply apis/cloudstack/src/main/java/org/jclouds/cloudstack/domain/UsageRecord.java]: @nacx Thanks for this, I noticed something when I was preparing the PR but never followed through.

I have removed usages of boolean and replaced with Boolean.

File: apis/cloudstack/src/main/java/org/jclouds/cloudstack/domain/UsageRecord.java [hunk 2]
```diff
@@ -476,7 +595,7 @@ public String getIpAddress() {
       return this.ipAddress;
    }
 
-   public boolean isSourceNAT() {
+   public Boolean isSourceNAT() {
```
Comment (nacx) [inline apis/cloudstack/src/main/java/org/jclouds/cloudstack/domain/UsageRecord.java]: Add the `@Nullable` annotation, for consistency.

**Comment by mcdonnell-john** [comment]
@nacx I think thats everything.

Thanks for the review

File: apis/cloudstack/src/main/java/org/jclouds/cloudstack/domain/UsageRecord.java (hunk 2 above)
Comment (nacx) [reply apis/cloudstack/src/main/java/org/jclouds/cloudstack/domain/UsageRecord.java]: @mcdonnell-john looks like this still needs to be addressed

Comment (mcdonnell-john) [reply apis/cloudstack/src/main/java/org/jclouds/cloudstack/domain/UsageRecord.java]: @nacx Your right, I only added the annotation to the constructor parameters. When I get home I will make this change and squash again.  Sorry about that.

Comment (mcdonnell-john) [reply apis/cloudstack/src/main/java/org/jclouds/cloudstack/domain/UsageRecord.java]: Fixed that there.

**Comment by nacx** [comment]
Merged to [master](http://git-wip-us.apache.org/repos/asf/jclouds/commit/25f2bcd8) and [2.1.x](http://git-wip-us.apache.org/repos/asf/jclouds/commit/a8b8c9dd). Thanks @mcdonnell-john!

**Pull request closed without merging**